tests/components/ecobee/test_config_flow.py

```diff
@@ -0,0 +1,63 @@
+"""Tests for the ecobee config flow."""
+from unittest.mock import patch
+
+from homeassistant import data_entry_flow
+from homeassistant.components.ecobee import config_flow
+from homeassistant.components.ecobee.const import DATA_ECOBEE_CONFIG
+from homeassistant.const import CONF_API_KEY
+
+
+async def test_abort_if_already_setup(hass):
+    """Test we abort if ecobee is already setup."""
+    flow = config_flow.EcobeeFlowHandler()
+    flow.hass = hass
+
+    with patch.object(hass.config_entries, "async_entries", return_value=[{}]):
+        result = await flow.async_step_user()
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "one_instance_only"
+
+
+async def test_user_step_without_user_input(hass):
+    """Test expected result if user step is called."""
+    flow = config_flow.EcobeeFlowHandler()
+    flow.hass = hass
+    flow.hass.data[DATA_ECOBEE_CONFIG] = {}
+
+    result = await flow.async_step_user()
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+
+async def test_pin_request_succeeds(hass):
+    """Test expected result if pin request succeeds."""
+    flow = config_flow.EcobeeFlowHandler()
+    flow.hass = hass
+    flow.hass.data[DATA_ECOBEE_CONFIG] = {}
+
+    with patch("homeassistant.components.ecobee.config_flow.Ecobee") as mock_ecobee:
+        mock_ecobee.request_pin.return_value = True
```

Comment:
Is it on the return value of `mock_ecobee` we should set the `request_pin` return value?

Reply:
That’s it! Pushed fixes. Still need to test import flow.